cpp/arcticdb/storage/storage_factory.cpp

```diff
@@ -54,6 +55,12 @@ std::shared_ptr<VariantStorageFactory> create_storage_factory(
         res = std::make_shared<VariantStorageFactory>(
             nfs_backed::NfsBackedStorageFactory(nfs_backed_config)
         );
+    } else if (type_name == azure::AzureStorage::Config::descriptor()->full_name()) {
+        azure::AzureStorage::Config azure_config;
+        storage.config().UnpackTo(&azure_config);
+        res = std::make_shared<VariantStorageFactory>(
+            azure::AzureStorageFactory(azure_config)
+        );
```

Comment:
It was probably fine when we had two storage types, but in a language with macros and templates, this level of manual copy and paste makes me sad.

Reply:
I have the same thought but do not have a good idea, as I think using macro here will sacrifice too much readability but not gaining too much. After all this part of the code is very straight forward. It should not need much maintenance  anyway. Pretty much write and forget.